### PR TITLE
docs: Readme documentation for the laser_encoder package

### PR DESCRIPTION
**Pull request by CaptainVee**
This commit adds comprehensive README documentation for the laser_encoders Python package. The README provides clear instructions on installation, primary usage, contributing guidelines, licensing information, etc.

### CODE REVIEW

**Review by heffernankevin — CHANGES_REQUESTED** [review]

File: laser_encoders/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+LASER Language-Agnostic SEntence Representations Toolkit
+
+laser_encoders is the official Python package for the Facebook [LASER](https://github.com/facebookresearch/LASER) library. It provides a simple and convenient way to use LASER embeddings in Python. It allows you to calculate multilingual sentence embeddings using the LASER toolkit. These embeddings can be utilized for various natural language processing tasks, including document classification, bitext filtering, and mining.
+
+## Dependencies
+
+- Python >= 3.8
+- [PyTorch](http://pytorch.org/)
+
```
[the remark this answers is not in the document]
Comment (CaptainVee) [reply laser_encoders/README.md:11]: Didn't know pipreqs before now. Thanks

File: laser_encoders/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+To tokenize a file use
+```py
+tokenized_sentence = tokenizer.tokenize_file(inp_fname=Path(path/to/input_file.txt), out_fname=Path(path/to/output_file.txt))
+```
+**Alternatively**, you can download and initialise the tokenizer and encoder with just one step
+```py
+from laser_encoders import initialize_encoder, initialize_tokenizer
+
+# Load the LASER tokenizer
+tokenizer = initialiize_tokenizer(lang="igbo")
```
[the remark this answers is not in the document]
Comment (CaptainVee) [reply laser_encoders/README.md]: fixed

[the remark this answers is not in the document]
Comment (CaptainVee) [reply laser_encoders/README.md]: Alright

[the remark this answers is not in the document]
Comment (CaptainVee) [reply laser_encoders/README.md]: done

File: laser_encoders/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
+# LASER encoders
+
+LASER Language-Agnostic SEntence Representations Toolkit
+
+laser_encoders is the official Python package for the Facebook [LASER](https://github.com/facebookresearch/LASER) library. It provides a simple and convenient way to use LASER embeddings in Python. It allows you to calculate multilingual sentence embeddings using the LASER toolkit. These embeddings can be utilized for various natural language processing tasks, including document classification, bitext filtering, and mining.
+
+## Dependencies
+
+- Python >= 3.8
```
Comment (heffernankevin) [inline laser_encoders/README.md:9]: Maybe we can just drop these and simply point users to `requirements.txt`.

File: laser_encoders/__main__.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import argparse
+
+from laser_encoders.download_models import download_models
+
+
+def main():
```
Comment (heffernankevin) [inline laser_encoders/__main__.py:6]: Why did you move the code from `download_models` here?

Comment (CaptainVee) [reply laser_encoders/__main__.py:6]: In other to be able to this
`python -m laser_encoders download_models`

with the way it was structured before, you had to run it like this
`python -m laser_encoders.download_models`

Comment (heffernankevin) [reply laser_encoders/__main__.py:6]: The original way seems find to me?

File: laser_encoders/README.md (hunk 3 above)
Comment (CaptainVee) [reply laser_encoders/README.md:9]: Alright, Though i feel having the major requirements right here give a better user experience

File: laser_encoders/__main__.py (hunk 4 above)
Comment (heffernankevin) [reply laser_encoders/__main__.py:6]: I would suggest simply reverting.

File: laser_encoders/README.md (hunk 3 above)
Comment (heffernankevin) [reply laser_encoders/README.md:9]: Okay, well if so we might as well be consistent. If you'd like to keep the major requirements here, then can you add in the suggested versions for each package (not just Python + PyTorch)

File: laser_encoders/__main__.py (hunk 4 above)
Comment (CaptainVee) [reply laser_encoders/__main__.py:6]: Alright.

File: laser_encoders/README.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+```sh
+ pip install laser_encoders
+```
+
+## Usage
+
+Here's a simple example of how you can download and initialise the tokenizer and encoder with just one step.
+
+**Note:** By default, the models will be downloaded to the` ~/.cache/laser_encoders` directory. To specify a different download location, you can provide the argument `model_dir=path/to/model/directory` to the initialize_tokenizer and initialize_encoder functions
```
Comment (heffernankevin) [inline laser_encoders/README.md:30]: [nit] remove extra space: ` ~/.cache/laser_encoders` -> `~/.cache/laser_encoders`

File: pyproject.toml [hunk 6]
```diff
@@ -19,7 +19,7 @@ dependencies = [
 ]
 
 classifiers=[
-    "License :: BSD License",
+    "License :: OSI Approved :: BSD License",
```
Comment (heffernankevin) [inline pyproject.toml:22]: Why did you add "OSI Approved"?

File: pyproject.toml [hunk 7]
```diff
@@ -7,7 +7,7 @@ name = "laser_encoders"
 version = "0.0.1"
 authors = [{name = "Facebook AI Research"}]
 description = "LASER  Language-Agnostic SEntence Representations is a toolkit to calculate multilingual sentence embeddings and to use them for document classification, bitext filtering and mining"
-readme = "README.md"
+readme = "laser_encoders/README.md"
```
Comment (heffernankevin) [inline pyproject.toml:10]: Is it necessary to prefix with `laser_encoders`?

File: laser_encoders/README.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+### Contributors
+
+The following people have contributed to this project:
+
+- [CaptainVee](https://github.com/CaptainVee)
+
+### License
+
+This package is released under the [LASER](https://github.com/facebookresearch/LASER/blob/main/LICENSE) Facebook BSD License.
```
Comment (heffernankevin) [inline laser_encoders/README.md:110]: [nit]: `Facebook BSD License` -> `BSD License`

File: laser_encoders/README.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+### Code of Conduct
+
+We expect all contributors to adhere to our [Code of Conduct](https://github.com/facebookresearch/LASER/blob/main/CODE_OF_CONDUCT.md).
+
+### Contributors
+
+The following people have contributed to this project:
+
+- [CaptainVee](https://github.com/CaptainVee)
```
Comment (heffernankevin) [inline laser_encoders/README.md:106]: The convention we usually use for contributors is to add everyone involved in alphabetical order (including those such as David and myself). However, as you are the core contributor, I would suggest that you keep your name at the top. Can you please also add: "Holger Schwenk" (@hoschwenk). This project is based off his core work and so he should also be added. You can remove the acknowledgements section (but very kind to have thought of it!)

**Review by heffernankevin — APPROVED** [review]

File: pyproject.toml (hunk 6 above)
Comment (CaptainVee) [reply pyproject.toml:22]: testpypi rejected it without the OSI. I searched online and found that so i added it and it worked

File: pyproject.toml (hunk 7 above)
Comment (CaptainVee) [reply pyproject.toml:10]: Yea, Still because of testpypi. If you don't point to it, it would use the README of the main laser repo which doesn't contain any information on how to run the laser package

Comment (heffernankevin) [reply pyproject.toml:10]: Do we not want the `pyproject.toml` file to live inside `laser_encoders`?

File: pyproject.toml (hunk 6 above)
Comment (heffernankevin) [reply pyproject.toml:22]: Ah ok! Thanks for checking!

File: pyproject.toml (hunk 7 above)
Comment (CaptainVee) [reply pyproject.toml:10]: Well I can't say for sure, you guys get to decide that. I worked based on the present structure. But my own suggestion too would be to move the `pyproject.toml` file

Comment (heffernankevin) [reply pyproject.toml:10]: Yes, let's move it into `laser_encoders`

Comment (CaptainVee) [reply pyproject.toml:10]: Alright then
